spec/features/favorites/index_spec.rb

```diff
@@ -62,5 +62,49 @@
 
       expect(page).not_to have_content("You haven't favorited any pets yet.")
     end
+
+    it "It sees a button to remove favorite next to each" do
+          visit "/pets/#{@pet_1.id}"
+
+          click_button 'Add to favorites'
+
+          visit "/pets/#{@pet_2.id}"
+
+          click_button 'Add to favorites'
+
+          visit "/favorites"
+
+          within "#pet-#{@pet_1.id}" do
+            expect(page).to have_button("Remove from favorites")
+          end
+
+          within "#pet-#{@pet_2.id}" do
+            expect(page).to have_button("Remove from favorites")
+          end
+    end
+
+    describe "Clicks a remove favorite button and the page is reloaded" do
+      it "It no longer sees that favorite on the page" do
+        visit "/pets/#{@pet_1.id}"
+
+
+        click_button 'Add to favorites'
+
+        visit "/pets/#{@pet_2.id}"
+
+        click_button 'Add to favorites'
+
+        visit "/favorites"
+
+        all(:button, 'Remove from favorites')[0].click
+
+        expect(page).to_not have_content("#{@pet_1.image}")
+        expect(page).to have_content("Favorite Pet Count: 1")
+
+        click_button 'Remove from favorites'
+        expect(page).to have_content("You haven't favorited any pets yet.")
+        expect(page).to have_content("Favorite Pet Count: 0")
+      end
+    end
```

Comment:
This all looks a bit messy to me but I'm not sure what to do about it, Ive asked both instructors about organizing these and both of them told me its just kind of up to the developer to decide so????

Reply:
One thing I like to do is make the describes kind of match up

```
RSpec.describe 'as a visitor' do
  describe 'when I visit my favorites index page' do
```

Reply:
I would like the tests to read something like 
As a visitor when I visit my favorites index page I can see my favorited pets names as links and images

describe 'as a visitor' do
describe 'when I visit my favorites index page'
it "can see my favorited pets names as links and images"
it "Sees no favorites message on index page user has no favorites"
it "Sees no no-favorites message on index page if user has favorites"
it "It sees a button to remove favorite next to each"
describe Clicks a remove favorite button and the page is reloaded
it "It no longer sees that favorite on the page"

Reply:
That does read better, I keep forgetting that the story is actually from the perspective of a user.

Reply:
Ok Ive made the changes to the describes.